### PR TITLE
Use non-deprecated code and remove unused imports

### PR DESCRIPTION
**Pull request by res0nance**
Removing all the deprecated code particularly in the WinRM bits

### CODE REVIEW

**Comment by res0nance** [comment]
I'm done with this, finally cleaned up most of the deprecated code

File: src/main/java/hudson/plugins/ec2/win/winrm/WinRMClient.java [hunk 1]
```diff
@@ -176,15 +183,40 @@ private void setupHTTPClient() {
...
+                                                            .build();
+            builder.setDefaultAuthSchemeRegistry(authSchemeRegistry);
         }
-        httpclient.setCredentialsProvider(credsProvider);
-        httpclient.getParams().setParameter(CoreConnectionPNames.CONNECTION_TIMEOUT, 5000);
-        // httpclient.setHttpRequestRetryHandler(new WinRMRetryHandler());
+        if (useHTTPS) {
+            try {
+                SSLConnectionSocketFactory sslConnectionFactory = new SSLConnectionSocketFactory(
+                                                                  new SSLContextBuilder().loadTrustMaterial(null, new TrustSelfSignedStrategy()).build(),
```
Comment (jvz) [inline src/main/java/hudson/plugins/ec2/win/winrm/WinRMClient.java]: Now I'm assuming this is a pure refactoring here, but this is not the greatest idea. There should probably be an admin-controlled config setting to disable hostname verification and whatnot.

Comment (res0nance) [reply src/main/java/hudson/plugins/ec2/win/winrm/WinRMClient.java]: This is pure refactoring, we should enhance this from a security perspective outside of this PR.

**Comment by res0nance** [comment]
I'll merge this tomorrow if no issues.

**Pull request merged**